assets/js/src/form_editor/store/reducers/history_record.ts

```diff
@@ -0,0 +1,83 @@
+import { isEqual } from 'lodash';
+
+export const createHistoryRecord = (state) => {
+  let editorHistory: any[] = state.editorHistory;
+  let editorHistoryOffset = state.editorHistoryOffset;
+  const lastHistoryRecord = editorHistory[editorHistory.length - 1];
+
+  let newHistoryRecord;
+  const time = Date.now();
+
+  if (
+    lastHistoryRecord !== undefined
+    && (time - lastHistoryRecord.time) < 1000
```

Comment:
Would it be better if we used a constants instead of using a magical numbers 1000 or 100 here?

Reply:
Yes, you are right. I'm not used to write constants in TS. I'll be better for next time.